src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NugetApiPackageManager.cs

```diff
@@ -234,27 +227,35 @@ private async Task<(PackageSource, IPackageSearchMetadata)> GetPackageMetadataAs
             _ = sources ?? throw new ArgumentNullException(nameof(sources));
 
             bool atLeastOneSourceValid = false;
-            foreach (PackageSource source in sources)
+
+            //internal cancellation token to cancel other tasks if matc
+            CancellationTokenSource internalCancelToken = new CancellationTokenSource();
+            using CancellationTokenSource linkedCts =
+                      CancellationTokenSource.CreateLinkedTokenSource(internalCancelToken.Token, cancellationToken);
```

Comment:
We don't need the `internalCancelToken` here. On the second line, using `CreateLinkedTokenSource(cancellationToken)` will already give us the ability to cancel "internally" using `linkedCts.Cancel()`.